pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -138,6 +142,10 @@ func (cl *Client) RegisterEventHandler(kind config.GroupVersionKind, handler mod
 
 // Run the queue and all informers. Callers should  wait for HasSynced() before depending on results.
 func (cl *Client) Run(stop <-chan struct{}) {
+	if cl.started.Load() {
```

Comment:
What is the impact if we do not do this? It looks racy, 2 threads can be waiting on L152 for a long time

Reply:
Yeah, make sense to me. 

Let me update

Reply:
Please let me know if we should also prevent run twice in the queue